packages/react-native-sdk/src/providers/StreamCall.tsx

```diff
@@ -134,3 +151,104 @@ const ClearPushWSSubscriptions = () => {
   }, []);
   return null;
 };
+
+/**
+ * This is a renderless component to get the device stats like thermal state and power saver mode.
+ */
+const DeviceStats = () => {
+  const { useCallCallingState } = useCallStateHooks();
+  const callingState = useCallCallingState();
+
+  const [lowPowerMode, setLowPowerMode] = useState<boolean | null>(null);
+  const [thermalState, setThermalState] = useState<string | null>(null);
+
+  const [powerModeSubscription, setPowerModeSubscription] =
+    useState<EmitterSubscription | null>(null);
+  const [thermalStateSubscription, setThermalStateSubscription] =
+    useState<EmitterSubscription | null>(null);
```

Comment:
subscriptions shouldnt be needed to be added into a state

get a subscription in useEffect and when returning clear it